### PR TITLE
chore: Download the `Cocoa SDK` instead of building it

### LINKED ISSUE #671: Reduce complexity of maintaining the `Cocoa SDK` plugin dependency
Building `sentry-cocoa` is currently part of the UE plugin [CI workflow](https://github.com/getsentry/sentry-unreal/blob/main/.github/workflows/ci.yml). In case of significant changes, regular updates of the corresponding submodule require additional effort to fix the existing build scripts for [iOS](https://github.com/getsentry/sentry-unreal/blob/main/scripts/build-ios.sh) and [Mac](https://github.com/getsentry/sentry-unreal/blob/main/scripts/build-mac.sh).

There are a few options to consider that can help to reduce the complexity of maintaining `Cocoa SDK` dependency:
- Download the required `sentry-cocoa` binaries from its GitHub release page ([example](https://github.com/getsentry/sentry-unity/pull/1854) of how this was done in Unity)
- Reuse the existing [build-xcframework.sh](https://github.com/getsentry/sentry-cocoa/blob/main/scripts/build-xcframework.sh) script which  `sentry-cocoa` relies on itself. Its current version is not optimized for Unreal (i.e. UE doesn't support Mac Catalyst or iPhone simulator) however it should be possible to introduce an extra parameter that will reduce build times and allow to skip unnecessary steps.

### PR DESCRIPTION
**Pull request by tustanivsky**
Closes #671 

#skip-changelog

### CODE REVIEW

**Review by bitsandfoxes — COMMENTED** [review]
Why are you splitting downloading iOS and macOS? It looks like you're taking them from the same downloaded xcframework?

**Comment by bitsandfoxes** [comment]
Could that not be a "prepare Cocoa" script/step? Instead of splitting it in two?

**Comment by tustanivsky** [comment]
I've merged `download-ios.sh` and `download-mac.sh` so that all cocoa-related preparations happen within a single script.

File: .github/workflows/update-dependencies.yml [hunk 1]
```diff
@@ -29,8 +29,8 @@ jobs:
   cocoa:
     uses: getsentry/github-workflows/.github/workflows/updater.yml@v2
     with:
-      path: modules/sentry-cocoa
-      name: Cocoa SDK (iOS)
+      path: modules/sentry-cocoa.properties
+      name: Cocoa SDK (iOS and Mac)
```
Comment (bruno-garcia) [inline .github/workflows/update-dependencies.yml:33]: Don't we use sentry-native on Mac?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,30 +6,22 @@
 
 - Assertions now print their callstack a log file ([#637](https://github.com/getsentry/sentry-unreal/pull/637))
 - Add UE 4.27, 5.0 and 5.1 to CI pipeline ([#675](https://github.com/getsentry/sentry-unreal/pull/675))
+- Download the `Cocoa SDK` instead of building it ([#681](https://github.com/getsentry/sentry-unreal/pull/681))
```
Comment (bruno-garcia) [inline CHANGELOG.md:9]: I don't think this needs to be in the changelog since for the end user, it doesn't make a difference, does it?

Comment (tustanivsky) [reply CHANGELOG.md]: You're right, this can be removed.

File: .github/workflows/update-dependencies.yml (hunk 1 above)
Comment (tustanivsky) [reply .github/workflows/update-dependencies.yml:33]: No, we switched to using sentry-cocoa on Mac quite a while ago (#190).

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**